common/Console.cc

```diff
@@ -14,19 +14,23 @@ static u8 row;
 
 void Init() {
     cols = VI::GetXFBWidth() / Font::GetGlyphWidth() - 1;
-    rows = VI::GetXFBHeight() / Font::GetGlyphHeight() - 1;
+    rows = VI::GetXFBHeight() / Font::GetGlyphHeight() / (!VI::IsProgressive() + 1) - 1;
     col = 0;
     row = 0;
 }
 
 static void Print(char c) {
-    if (col >= cols) {
+    if (col > cols && c != '\n' && c != '\r') {
         return;
     }
 
     if (c == '\n') {
         col = 0;
-        row++;
+
+        if (row++ == rows) {
+            row = 0;
+        }
+
         return;
```

Comment:
```suggestion
        row++;

        return;
```
This break things e.g. if `Console::Print("\n\n")` is called while the console is already full. I don't think that was your intent here.

Reply:
Corrected in commit [6ff7b0b](https://github.com/stblr/mkw-sp/pull/670/commits/6ff7b0b67a312fa1966f90a93c6c7aa21ab251d5).